recipes/gdcm/all/conanfile.py

```diff
@@ -0,0 +1,219 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import textwrap
+
+required_conan_version = ">=1.33.0"
+
+class GDCMConan(ConanFile):
+    name = "gdcm"
+    topics = ("dicom", "images")
+    homepage = "http://gdcm.sourceforge.net/"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSD-3-Clause"
+    description = "C++ library for DICOM medical files"
+    exports_sources = "CMakeLists.txt", "patches/**"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if (self.settings.compiler == "Visual Studio"
+                and self.options.shared
+                and str(self.settings.compiler.runtime).startswith("MT")):
+            raise ConanInvalidConfiguration("shared gdcm can't be built with MT or MTd")
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "11")
+
+    def requirements(self):
+        self.requires("expat/2.4.1")
+        self.requires("openjpeg/2.4.0")
+        self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["GDCM_BUILD_DOCBOOK_MANPAGES"] = False
+        self._cmake.definitions["GDCM_BUILD_SHARED_LIBS"] = self.options.shared
+        self._cmake.definitions["GDCM_USE_SYSTEM_EXPAT"] = True
+        self._cmake.definitions["GDCM_USE_SYSTEM_OPENJPEG"] = True
+        self._cmake.definitions["GDCM_USE_SYSTEM_ZLIB"] = True
```

Comment:
What about all these others options to use more external dependencies: https://github.com/malaterre/GDCM/blob/17dc99ed77f10aada90448c524fcd0be62a40365/CMakeLists.txt#L313-L345

Reply:
Yuck that's a good catch! 

https://github.com/malaterre/GDCM/blob/17dc99ed77f10aada90448c524fcd0be62a40365/CMakeLists.txt#L421

I think "unvendoring" can be for a separate PR

Reply:
- libuuid probably can be integrated but requires more patching
- socketxx isn't present in cci
- charls: conan-center has only charls 2.1.0. gdcm isn't compatible with it. https://sourceforge.net/p/gdcm/bugs/505/
- libjpeg has been changed to be able to build and use simultaneously 3 variants of library (8/12/16 bits)
- json-c, openssl, libxml2, poppler, papyrus3 aren't used for building libraries

Reply:
ok, so could you add a TODO to eventually unvendor libuuid, socketxx & charls? Otherwise recipe LGTM.

Reply:
```suggestion
        # FIXME: unvendor deps https://github.com/conan-io/conan-center-index/pull/5705#discussion_r647224146
        self._cmake.definitions["GDCM_USE_SYSTEM_EXPAT"] = True
        self._cmake.definitions["GDCM_USE_SYSTEM_OPENJPEG"] = True
        self._cmake.definitions["GDCM_USE_SYSTEM_ZLIB"] = True
```